api/src/test/java/org/openmrs/api/ProviderServiceTest.java

```diff
@@ -478,12 +478,32 @@ public void getCountOfProviders_shouldFetchNumberOfProviderMatchingGivenQuery()
 		Context.getPersonService().savePerson(person);
 		assertEquals(1, service.getCountOfProviders("Hippo").intValue());
 	}
+
+	@Test
+	public void getCountOfProviders_shouldHandleMoreThanOneMatch() {
+	  
+	  // creating one more matching provider
+	  Person person = Context.getPersonService().getPerson(42);
+	  Set<PersonName> names = person.getNames();
+	  for (PersonName name : names) {
+	    name.setVoided(true);
+	  }
+	  PersonName personName = new PersonName("hippopotamus", "foo", "bar");
+	  personName.setPreferred(true);
+	  person.addName(personName);
+	  person = Context.getPersonService().savePerson(person);
+	  Provider provider = new Provider();
+	  provider.setPerson(person);
+	  Context.getProviderService().saveProvider(provider);
+	  
+	  // verif
+	  assertEquals(2, service.getCountOfProviders("Hippo").intValue());
+	}
```

Comment:
@samuelmale, this test was only added to demonstrate the bug. Now that you have re-enabled the two tests that were `@Ignore`'d, this one can go away.

Reply:
Yes, I was gonna do that before. Let me make more changes :)